src/components/AnnouncementBanner/AnnouncementBanner.tsx

```diff
@@ -31,6 +37,47 @@ export const AnnouncementBanner: FC<AnnouncementBannerProps> = ({
     return null;
   }
 
+  const handleRewriteLinkBehavior = (e: React.MouseEvent<HTMLElement>) => {
+    if ((e.target as HTMLElement)?.tagName?.toLowerCase() === 'a') {
+      e.preventDefault();
+      e.stopPropagation();
+
+      const link = (e.target as HTMLElement).closest('a') as HTMLAnchorElement;
+      const href = link.getAttribute('href');
+
+      if (!href) {
+        return;
+      }
+
+      if (isExternalUrl(href)) {
+        openInNewTab(href);
+        return;
+      }
+
+      if (/fromChain|toChain|fromToken|toToken/g.test(href)) {
+        const url = new URL(href, window.location.origin);
+        const searchParams = url.searchParams;
+        if (searchParams.get('fromChain')) {
+          widgetCache.setFromChainId(
+            parseInt(searchParams.get('fromChain') ?? ''),
+          );
+        }
+        if (searchParams.get('fromToken')) {
+          widgetCache.setFromToken(searchParams.get('fromToken') ?? '');
+        }
+        if (searchParams.get('toChain')) {
+          widgetCache.setToChainId(parseInt(searchParams.get('toChain') ?? ''));
+        }
+        if (searchParams.get('toToken')) {
+          widgetCache.setToToken(searchParams.get('toToken') ?? '');
+        }
+        return;
+      }
+
+      router.push(href);
+    }
+  };
+
```

Comment:
This is needed to override the `a` default behaviour so that we'll keep a single source of truth for the widget params in widgetCache, which we use also for the wallet menu navigation